web/src/test/auto/headless/engine/predictive-text/worker-thread/correction-search/classical-calculation.tests.ts

```diff
@@ -459,86 +459,199 @@ describe('Classical Damerau-Levenshtein edit-distance calculation', function() {
-      assert.deepEqual(buffer.editPath(), editSequence);
+      const viablePaths = buffer.editPath();
+      assert.equal(viablePaths.length, 1);
+      assert.deepEqual(viablePaths[0], editSequence);
     });
 
     it("'ifhs' -> 'fish'", function() {
       let buffer = compute("ifhs", "fish");
 
-      let editSequence = [
-          op.ts, op.te, op.ts, op.te
+      let editSequence: EditTuple<string>[] = [
+        { op: 'transpose-start', input: 'i', match: 'f'},
+        { op: 'transpose-end', input: 'f', match: 'i'},
+        { op: 'transpose-start', input: 'h', match: 's'},
+        { op: 'transpose-end', input: 's', match: 'h'}
       ];
 
-      assert.deepEqual(buffer.editPath(), editSequence);
+      const viablePaths = buffer.editPath();
+      assert.equal(viablePaths.length, 1);
+      assert.deepEqual(viablePaths[0], editSequence);
     });
 
     it("'then' -> 'their'", function() {
       let buffer = compute("then", "their");
 
-      let editSequence = [
-          op.m, op.m, op.m, op.s, op.i
+      let editSequence: EditTuple<string>[] = [
+        { op: 'match', input: 't', match: 't'},
+        { op: 'match', input: 'h', match: 'h'}, // best sequence:  continguous matches
+        { op: 'match', input: 'e', match: 'e'}, // + adjacent substitutes
+        { op: 'substitute', input: 'n', match: 'i'},
+        { op: 'insert', match: 'r'},
       ];
 
-      assert.deepEqual(buffer.editPath(), editSequence);
+      let altSequence: EditTuple<string>[] = [
+        { op: 'match', input: 't', match: 't'},
+        { op: 'match', input: 'h', match: 'h'},
+        { op: 'match', input: 'e', match: 'e'},
+        { op: 'insert', match: 'i'},
+        { op: 'substitute', input: 'n', match: 'r'},
+      ];
+
+      const viablePaths = buffer.editPath();
+      assert.equal(viablePaths.length, 2);
+      assert.deepEqual(viablePaths[0], editSequence);
+      assert.deepEqual(viablePaths[1], altSequence);
     });
 
     // Two transpositions:  abc -> ca, ig <- ghi.  Also, one deletion:  'd'.
     it("'abczdefig' -> 'cazefghi'", function() {
       let buffer = compute("abczdefig", "cazefghi", "InputThenMatch", 2);
 
-      let editSequence = [
-        // --- abc -> ca ----
-        // a/c    b/_    c/a    z     -d    e     f
-          op.ts, op.td, op.te, op.m, op.d, op.m, op.m,
-        // --- ig -> ghi ----
-        // i/g    _/h    g/i
-          op.ts, op.ti, op.te
+      let editSequence: EditTuple<string>[] = [
+        { op: 'transpose-start', input: 'a', match: 'c'},
+        { op: 'transpose-delete', input: 'b'},
+        { op: 'transpose-end', input: 'c', match: 'a'},
+        { op: 'match', input: 'z', match: 'z'},
+        { op: 'delete', input: 'd'},
+        { op: 'match', input: 'e', match: 'e'},
+        { op: 'match', input: 'f', match: 'f'},
+        { op: 'transpose-start', input: 'i', match: 'g'},
+        { op: 'transpose-insert', match: 'h'},
+        { op: 'transpose-end', input: 'g', match: 'i'}
+      ];
```

Comment:
Visualized:

```
    cazefghi

  0|12
  ----------
a 1|112      transpose-start('a' vs 'c')
b 2|2223     transpose-delete('b')
c  |22334    transpose-end('c' vs 'a')
z  | 32345   match('z' == 'z')
d  |  33456  delete('d')
e  |   34567 match('e' == 'e')
f  |    3456 match('f' == 'f')
i  |     455 transpose-start('i' vs 'g')
g  |      55 transpose-insert('h'), transpose-end('g' vs 'i')
```

Note that there are 7 total non-`match` entries, but the 2 sets of `transpose-` entries each count for one less edit.  (Set of 3 => 2 edit cost)  That gives us the final total value of 5, which is the count seen in the bottom-most entry in the number table.

The test's diagonal width should technically be bigger here, but it works out for these particular values anyway.